### PR TITLE
add v8 to ci

### LINKED ISSUE #68: Update tests to V8 VSPG
Since 2023-06-02 for version 8.18.1 of Exasol database a Docker image is available on [Dockerhub](https://hub.docker.com/layers/exasol/docker-db/8.18.1/images/sha256-33ce379848b9c2d3c0535af46a9e23382f1d3ccf5fc8c27131ba5c3e6e19a065).

The current ticket therefore requests to update the integration tests of VSPG to use version 8.18.1 as latest default version.

Please note sibling-tickets for all [JDBC-based virtual schemas](https://github.com/exasol/virtual-schemas/blob/main/doc/user-guide/dialects.md).

### LINKED ISSUE #78: Fix vulnerabilities in org.postgresql:postgresql:jar:42.6.0:compile & org.apache.commons:commons-compress:jar:1.24.0:test
```
 Error:  Failed to execute goal org.sonatype.ossindex.maven:ossindex-maven-plugin:3.2.0:audit (default-cli) on project postgresql-virtual-schema: Detected 2 vulnerable components:
Error:    org.apache.commons:commons-compress:jar:1.24.0:test; https://ossindex.sonatype.org/component/pkg:maven/org.apache.commons/commons-compress@1.24.0?utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:      * [CVE-2024-25710] CWE-835: Loop with Unreachable Exit Condition ('Infinite Loop') (8.1); https://ossindex.sonatype.org/vulnerability/CVE-2024-25710?component-type=maven&component-name=org.apache.commons%2Fcommons-compress&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:      * [CVE-2024-26308] CWE-770: Allocation of Resources Without Limits or Throttling (7.5); https://ossindex.sonatype.org/vulnerability/CVE-2024-26308?component-type=maven&component-name=org.apache.commons%2Fcommons-compress&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:    org.postgresql:postgresql:jar:42.6.0:compile; https://ossindex.sonatype.org/component/pkg:maven/org.postgresql/postgresql@42.6.0?utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:      * [CVE-2024-1597] CWE-89: Improper Neutralization of Special Elements used in an SQL Command ('SQL Injection') (10.0); https://ossindex.sonatype.org/vulnerability/CVE-2024-1597?component-type=maven&component-name=org.postgresql%2Fpostgresql&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
```

### PR DESCRIPTION
**Pull request by pj-spoelders**
Closes #68 
Closes #78 

The behaviour when it comes to character sets is now simplified,
The target char set is now always UTF-8.
The `IMPORT_DATA_TYPES` property (and value `FROM_RESULT_SET`) are now deprecated (change in vs-common-jdbc):
An exception will be thrown when users use `FROM_RESULT_SET`. The exception message warns the user that the value is no longer supported and the property itself is also deprecated.

Various broken scalar time-related extraction functions for dates and timestamps in the virtual schema are now fixed: `year`,`month`,`day`,`hour`,`minute`,`second`.

Scalar `division` (`/`) which was broken in some cases now also works correctly.

Tests for `current_schema` are currently disabled, this is because of a discovered compiler bug: https://github.com/exasol/postgresql-virtual-schema/issues/79 . 
These tests will be re-evaluated later when there is more clarity about this issue.

We also updated dependencies and resolved the following 2 CVEs in test dependency `org.apache.commons:commons-compress`:
- CVE-2024-26308
- CVE-2024-25710
We also updated dependencies and resolved the following CVE in test dependency `org.postgresql:postgresql:`:
- CVE-2024-1597

### CODE REVIEW

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java [hunk 1]
```diff
@@ -86,10 +95,12 @@ public class PostgresVirtualSchemaIntegrationTestSetup implements Closeable {
     private static void uploadDriverToBucket(final Bucket bucket)
             throws InterruptedException, TimeoutException, BucketAccessException {
         try {
-            bucket.uploadFile(JDBC_DRIVER_PATH, JDBC_DRIVER_NAME);
+            bucket.uploadStringContent(JDBC_DRIVER_CONFIGURATION_FILE_CONTENT,
+                    "drivers/jdbc/" + JDBC_DRIVER_CONFIGURATION_FILE_NAME);
+            bucket.uploadFile(JDBC_DRIVER_PATH, "drivers/jdbc/" + JDBC_DRIVER_NAME);
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java:100]: Please replace this with the [driver manager](https://github.com/exasol/exasol-testcontainers/blob/main/doc/user_guide/user_guide.md#installing-drivers-for-external-data-sources-and-sinks) from exasol-testcontainers.

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java [hunk 2]
```diff
@@ -86,10 +95,12 @@ public class PostgresVirtualSchemaIntegrationTestSetup implements Closeable {
...
         } catch (final BucketAccessException | FileNotFoundException exception) {
             throw new IllegalStateException(
-                    ExaError.messageBuilder("F-VSPG-8")
+                    ExaError.messageBuilder("F-VSMYSQL-2")
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java:103]:
```suggestion
                    ExaError.messageBuilder("F-VSPG-8")
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java (hunk 1 above)
Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java]: done!

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java (hunk 2 above)
Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java]: right ..corrected

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java [hunk 3]
```diff
@@ -83,18 +86,25 @@ public class PostgresVirtualSchemaIntegrationTestSetup implements Closeable {
...
-        } catch (final BucketAccessException | FileNotFoundException exception) {
+            container.getDriverManager().install( //
+                    JdbcDriver.builder("POSTGRES_JDBC_DRIVER") //
+                            .enableSecurityManager(false) //
+                            .mainClass("org.postgresql.Driver") //
+                            .prefix("jdbc:postgresql:") //
+                            .sourceFile(JDBC_DRIVER_PATH) //
+                            .build());
+
+        } catch (final Exception ex) {
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java:100]:
```suggestion
        } catch (final Exception exception) {
```

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java [hunk 4]
```diff
@@ -83,18 +86,25 @@ public class PostgresVirtualSchemaIntegrationTestSetup implements Closeable {
...
             throw new IllegalStateException(
                     ExaError.messageBuilder("F-VSPG-8")
                             .message("An error occurred while uploading the jdbc driver to the bucket.")
                             .mitigation("Make sure the {{JDBC_DRIVER_PATH}} file exists.")
                             .parameter("JDBC_DRIVER_PATH", JDBC_DRIVER_PATH)
                             .mitigation("You can generate it by executing the integration test with maven.").toString(),
-                    exception);
+                    ex);
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java:107]:
```suggestion
                    exception);
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/adapter/dialects/postgresql/PostgresSQLSqlGenerationVisitor.java [hunk 5]
```diff
@@ -129,7 +136,22 @@ private String getDateTime(final List<String> argumentsSql, final ScalarFunction
...
+        case WEEK:
+        case MONTH:
+            builder.append("2");
+            break;
+        case YEAR:
+            builder.append("4");
+            break;
+        default:
+            break;
+        }
```
[the remark this answers is not in the document]
Comment (pj-spoelders) [reply src/main/java/com/exasol/adapter/dialects/postgresql/PostgresSQLSqlGenerationVisitor.java:153]: Ok

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java (hunk 3 above)
Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java:100]: Will modify it

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java (hunk 4 above)
Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/postgresql/PostgresVirtualSchemaIntegrationTestSetup.java:107]: Will modify it

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgresSQLSqlGenerationVisitorTest.java [hunk 6]
```diff
@@ -29,90 +29,92 @@
...
-    private PostgresSQLSqlGenerationVisitor visitor;
+        private PostgresSQLSqlGenerationVisitor visitor;
 
-    @BeforeEach
-    void beforeEach(@Mock final ConnectionFactory connectionFactoryMock) {
-        final SqlDialect dialect = new PostgreSQLSqlDialect(connectionFactoryMock, AdapterProperties.emptyProperties());
-        final SqlGenerationContext context = new SqlGenerationContext("test_catalog", "test_schema", false);
-        this.visitor = new PostgresSQLSqlGenerationVisitor(dialect, context);
-    }
+        @BeforeEach
```
[the remark this answers is not in the document]
Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/postgresql/PostgresSQLSqlGenerationVisitorTest.java:34]: Seems OK?

File: .github/workflows/dependencies_update.yml
[the remark this answers is not in the document]
Comment (pj-spoelders) [reply .github/workflows/dependencies_update.yml:1]: Done!

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgresSQLSqlGenerationVisitorTest.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/postgresql/PostgresSQLSqlGenerationVisitorTest.java:34]: Oh I see what you mean, let me fix that.

[the remark this answers is not in the document]
Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/postgresql/PostgresSQLSqlGenerationVisitorTest.java]: Should be fixed.

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**